src/lib/coda_base/coinbase.mli

```diff
@@ -44,7 +44,7 @@ val supply_increase : t -> Currency.Amount.t Or_error.t
 val fee_excess : t -> Currency.Fee.Signed.t Or_error.t
 
 module Gen : sig
-  val gen : t Quickcheck.Generator.t
+  val gen : coinbase_amount:Currency.Amount.t -> t Quickcheck.Generator.t
```

Comment:
wonder if this should have a different name, because `gen` usually is the generator, without any initialization parameters

`gen_of_coinbase_amount` ?

Reply:
Added a comment, but left the name. I think it's probably okay to call it `gen` if the arguments are to pass global configuration, even if it does break the pattern a bit.

Reply:
we do have other `gen`s in the codebase that take arguments so maybe we can go and fix them all at once later if we decide to.